src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -745,23 +745,23 @@ void SpatialPooler::bumpUpWeakColumns_() {
 
 
 void SpatialPooler::updateDutyCyclesHelper_(vector<Real> &dutyCycles,
-                                            SDR &newValues,
-                                            UInt period) {
+                                            const SDR &newValues,
+                                            const UInt period) {
   NTA_ASSERT(period > 0);
-  NTA_ASSERT(dutyCycles.size() == newValues.size);
+  NTA_ASSERT(dutyCycles.size() == newValues.getSparse().size()) << "duty dims: " << dutyCycles.size() << " SDR dims: " << newValues.size;
```

Comment:
I think that this check should not have `getSparse`, instead it is supposed to check the SDR size, like it did before.